packages/x-data-grid-pro/src/hooks/features/rows/useGridRowAriaAttributes.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+import {
+  GridRowId,
+  useGridSelector,
+  gridFilteredTopLevelRowCountSelector,
+  gridFilteredChildrenCountLookupSelector,
+  gridExpandedSortedRowIdsLookupSelector,
+  GRID_ROOT_GROUP_ID,
+} from '@mui/x-data-grid';
+import { useGridRowAriaAttributes as useGridRowAriaAttributesCommunity } from '@mui/x-data-grid/internals';
+import { useGridPrivateApiContext } from '../../utils/useGridPrivateApiContext';
+import { useGridRootProps } from '../../utils/useGridRootProps';
+
+export const useGridRowAriaAttributes = () => {
+  const apiRef = useGridPrivateApiContext();
+  const props = useGridRootProps();
+  const getRowAriaAttributesCommunity = useGridRowAriaAttributesCommunity();
+
+  const filteredTopLevelRowCount = useGridSelector(apiRef, gridFilteredTopLevelRowCountSelector);
+  const filteredChildrenCountLookup = useGridSelector(
+    apiRef,
+    gridFilteredChildrenCountLookupSelector,
+  );
+  const sortedVisibleRowPositionsLookup = useGridSelector(
+    apiRef,
+    gridExpandedSortedRowIdsLookupSelector,
+  );
+
+  return React.useCallback(
+    (rowId: GridRowId, index: number) => {
+      const rowNode = apiRef.current.getRowNode(rowId);
+      const ariaAttributes = getRowAriaAttributesCommunity(rowId, index);
+
+      if (rowNode === null || props.treeData !== true) {
```

Comment:
This might be a topic for a follow-up PR, but should we reuse the same ARIA attributes for the row grouping as well?
Looking at https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/treegrid_role, I think we should also set `role="treegrid"` when row grouping is enabled.

Reply:
I have added it in this PR